pkg/cloudprovider/providers/aws/cloudformation_test.go

```diff
@@ -0,0 +1,136 @@
+package aws
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/UKHomeOffice/keto/pkg/cloudprovider/providers/aws/mocks"
+	"github.com/UKHomeOffice/keto/pkg/model"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+)
+
+func TestDescribeStacks(t *testing.T) {
+	mockCF := &mocks.CloudFormationAPI{}
+	c := &Cloud{
+		cf: mockCF,
+	}
+
+	mockCF.On("DescribeStacks", &cloudformation.DescribeStacksInput{}).Return(
+		&cloudformation.DescribeStacksOutput{}, awserr.New("ValidationError", "does not exist", nil),
+	)
+
+	res, err := c.describeStacks("")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(res) != 0 {
+		t.Errorf("got number of results: %d; want: %d", len(res), 0)
+	}
+	if !mockCF.AssertExpectations(t) {
+		t.Errorf("not everything got called as expected")
+	}
+}
+
+func TestGetStacksByType(t *testing.T) {
+	mockCF := &mocks.CloudFormationAPI{}
+	c := &Cloud{
+		cf: mockCF,
+	}
+
+	stacks := []*cloudformation.Stack{
```

Comment:
This datastructure is begging to have a factory to generate test instances (with sensible defaults), given how much you create it (but this can be done at a later date). See something like http://www.rubydoc.info/gems/factory_girl/file/GETTING_STARTED.md for what I mean (or happy to describe in RIL).